R/clean_redcap_long.R

```diff
@@ -399,17 +398,37 @@ distill_repeat_table_long <- function(form_name,
 
 convert_mixed_instrument <- function(db_data_long, mixed_structure_ref) {
   for (i in seq_len(nrow(mixed_structure_ref))) {
-    field <- mixed_structure_ref$field_name[i]
-    form <- mixed_structure_ref$form_name[i]
-
-    # Create a logical mask for rows needing update
-    update_mask <- is.na(db_data_long$redcap_repeat_instance) & !is.na(db_data_long[[field]])
+    field <- mixed_structure_ref$field_name[i] # nolint: object_usage_linter
+    form <- mixed_structure_ref$form_name[i] # nolint: object_usage_linter
 
-    # Update redcap_repeat_instance
-    db_data_long$redcap_repeat_instance <- if_else(update_mask, 1, db_data_long$redcap_repeat_instance)
+    # Create an update mask column to identify which mixed structure rows need updates
+    db_data_long <- db_data_long %>%
+      mutate(
+        update_mask = case_when(
+          # repeat separately instances
+          !is.na({{ field }}) & is.na(.data$redcap_repeat_instance) ~ TRUE,
```

Comment:
I don't think `!is.na({{ field }})` is actually working. It always returns `TRUE`. Some NSE and `tidyselect` alternatives:

```r
library(tidyverse)

col <- "x"

tibble(x = c(1, NA)) |>
  mutate(y = if_else(is.na({{col}}), 1, 0))

#># A tibble: 2 × 2
#>      x     y
#>  <dbl> <dbl>
#>1     1     0
#>2    NA     0

# Wrong

tibble(x = c(1, NA)) |>
  mutate(y = if_else(is.na(!!as.symbol(col)), 1, 0))

#># A tibble: 2 × 2
#>      x     y
#>  <dbl> <dbl>
#>1     1     0
#>2    NA     1

# Right

tibble(x = c(1, NA)) |>
  mutate(across(all_of(col), ~if_else(is.na(.), 1, 0), .names = "y"))

#># A tibble: 2 × 2
#>      x     y
#>  <dbl> <dbl>
#>1     1     0
#>2    NA     1

# Right
```

Reply:
Ah thanks for the great reprex. I'll go for option 2 for now unless you have a strong preference.